helpers/PoseidonDeployHelper.ts

```diff
@@ -29,13 +28,35 @@ export async function deployPoseidons(
   });
 
   const deployPoseidon = async (params: number) => {
-    const abi = poseidonContract.generateABI(params);
-    const code = poseidonContract.createCode(params);
-    const PoseidonElements = new ethers.ContractFactory(abi, code, deployer);
-    const poseidonElements = await PoseidonElements.deploy();
-    await poseidonElements.waitForDeployment();
-    console.log(`Poseidon${params}Elements deployed to:`, await poseidonElements.getAddress());
-    return poseidonElements;
+    let poseidonModule: any;
+    switch(params) {
+      case 1:
+        poseidonModule = Poseidon1Module;
+        break;
+      case 2:
+        poseidonModule = Poseidon2Module;
+        break;
+      case 3:
+        poseidonModule = Poseidon3Module;
+        break;
+      case 4:
+        poseidonModule = Poseidon4Module;
+        break;
+      case 5:
+        poseidonModule = Poseidon5Module;
+        break;
+      case 6:
+        poseidonModule = Poseidon6Module;
+        break;
+    }
+    
+    const poseidonDeploy = await ignition.deploy(poseidonModule, {
```

Comment:
if I don't use create2 will regular deployment work through ignition?

Reply:
yes

Reply:
ignition deploy has optional param `strategy`, 
it can take 2 options:
```
export interface StrategyConfig {
    basic: Record<PropertyKey, never>;
    create2: {
        salt: string;
    };
}
```
if `create2` is used, that we should pass `salt`, or have one in hardhat config (we have in config)